src/schemas/components/SchemaFormPropertyAllOf.vue

```diff
@@ -33,11 +33,15 @@
 
   const schema = useSchema()
 
-  const property = computed(() => props.property.allOf.reduce<SchemaProperty>((property, definition) => {
-    if (isPropertyWith(definition, '$ref')) {
-      return merge(property, getSchemaDefinition(schema, definition.$ref))
-    }
+  const property = computed(() => {
+    const definitions = props.property.allOf.reduce<SchemaProperty>((property, definition) => {
+      if (isPropertyWith(definition, '$ref')) {
+        return merge(getSchemaDefinition(schema, definition.$ref), property)
+      }
 
-    return merge(property, definition)
-  }, {}))
+      return merge(property, definition)
+    }, {})
+
+    return merge(definitions, props.property)
```

Comment:
Noticed that the original title was being overridden by any title from the references. Merging the original property over the references makes sure the values on the property itself always win.